frozen-abi/src/abi_example.rs

```diff
@@ -521,9 +526,17 @@ impl<T: Serialize + IgnoreAsHelper> AbiEnumVisitor for &T {
 // inability of implementing AbiExample for private structs from other crates
 impl<T: Serialize + IgnoreAsHelper + EvenAsOpaque> AbiEnumVisitor for &T {
     default fn visit_for_abi(&self, digester: &mut AbiDigester) -> DigestResult {
-        info!("AbiEnumVisitor for (IgnoreAsOpaque): {}", type_name::<T>());
-        let top_scope = type_name::<T>().split("::").next().unwrap();
-        self.serialize(digester.create_new_opaque(top_scope))
+        let type_name = type_name::<T>();
+        let scope = if let Some(scope) = T::SCOPE {
+            scope
+        } else {
+            type_name.split("::").next().unwrap()
+        };
```

Comment:
Not sure I understand this works.

By default `EvenAsOpaque` will have `None` for `SCOPE`. In which case, we would split the `type_name` on `::`.
What does `type_name` include? If we're splitting on `::` would this result be the module it's in?

The only place I see `SCOPE` being set is for `PacketFlags` where the scope would be some "InternalBitFlags".
Why do we need to replace the scope there?

Reply:
those are good questions! hope this commit helps your understanding: 95b229fba6cc86249e83e91c39cceb897625c258